environs/bootstrap/config.go

```diff
@@ -288,6 +306,20 @@ func NewConfig(attrs map[string]interface{}) (Config, error) {
 		config.CAPrivateKey = caKeyPem
 	}
 
+	// Try get the key from the attribute map or try load it from file.
+	// If it isn't present, this is OK, we allow the controller to generate it
+	// on "jujud bootstrap-state" command.
```

Comment:
I see for the CA cert above, if it is not provided, one is generated here. Why not do the same for the host key? Is that because this is on the client, and we want to generate on the controller?

Reply:
We needed the CACert at the current call site to populate a controller config struct. We don't need the ssh host key in the same way. The original code for all this is years old. We'd do it differently now.